test/unit/init/init_test.js

```diff
@@ -87,15 +89,74 @@ suite('init', () => {
   suite('promptGeneratorSelection', () => {
 
     let yeomanEnvMock;
+    const GENERATORS = [
+      {
+        generatorName: 'polymer-init-element:app',
+        metaName: 'polymer-init-element',
+        shortName: 'element',
+        description: 'A blank element template',
+        resolved: 'unknown',
+      },
+      {
+        generatorName: 'polymer-init-my-test-app:app',
+        metaName: 'polymer-init-my-test-app',
+        shortName: 'my-test-app',
+        description: 'my test app',
+      },
+      {
+        generatorName: 'polymer-init-polymer-starter-kit-custom-1:app',
+        metaName: 'polymer-init-polymer-starter-kit-psk-1',
+        shortName: 'psk-1',
+        description: 'PSK 1',
+      },
+      {
+        generatorName: 'polymer-init-polymer-starter-kit-custom-2:app',
+        metaName: 'generator-polymer-init-polymer-starter-kit-psk-2',
+        shortName: 'psk-2',
+        description: 'PSK 2',
+      },
+      {
+        generatorName: 'polymer-init-custom-build-1:app',
+        metaName: 'generator-polymer-init-custom-build-1',
+        shortName: 'custom-build-1',
+        description: 'custom build 1',
+      },
+      {
+        generatorName: 'polymer-init-custom-build-2:app',
+        metaName: 'polymer-init-custom-build-2',
+        shortName: 'custom-build-2',
+        description: 'custom build 2',
+      },
+      {
+        generatorName: 'polymer-init-custom-build-3:app',
+        metaName: 'custom-build-3',
+        shortName: 'custom-build-3',
+        description: 'custom build 3',
+      },
+    ];
 
     setup(() => {
+      let generators = {};
+
+      for (let g of GENERATORS) {
+        if (!g.resolved) {
+          let tmpDir = temp.mkdirSync();
+          let packageJsonPath = `${tmpDir}/package.json`;
+          fs.writeFileSync(packageJsonPath, JSON.stringify({
```

Comment:
Yep, still needed. It's actually `polymer-cli` that's checking. [`getGeneratorMeta()`](https://github.com/Polymer/polymer-cli/pull/472/files#diff-6eb970fedd99f33facb11eefde800f22R119) reads the `package.json` from the directory indicated by the generator's `resolved` attribute. I'm not sure how to mock that out, but let me know.

(The link for `getGeneratorMeta()` jumps to the relevant line (119), but it's hidden underneath the topbar, so you'll have to scroll a bit to see the line :octocat: :bug:).